articles/building-a-simple-cryptocurrency-blockchain/index.md

```diff
@@ -0,0 +1,187 @@
+        }   
+    }
+```js
+
+I'll explain each part of the code here:
+
+I've created the class BlockCrytpo for my ***block*** and added a constructor just like any other JavaScript class. 
+
+In the constructor we initialize its properties and Assign parameters to it as follows:
+
+    
+
+- `crypto-js/sha256`: This is the module we've imported to calculate the hash of each block and we convert it to string using 'toString' method as the module will return the object.
+
+- `index`:  This is a distinctive number tracking the index of every block in the blockchain
+
+- `current_time`: As the name states, it keeps a record of the time when each transaction is completed. 
+
+- `info`: Keeps record and provide data about completed transactions 
+
+- `nexthash` : It points to the hash of the next block in the chain, used to keep and maintain the integrity of the blockchain.
+
+- `computeHash` We use this method to calculate the hash of the next block based on its properties.
+
+### The Blockchain Theorem
+
+The **blockchain Theorem** is core. This is where all blocks are connected. 
+
+So let's create a class that will maintain this operation.
+
+Here is the code for the class I named 'Blockchain'
+
+```Node
+    class Blockchain{
+        constructor(){
+            this.block1chain = [this.startGenesisBlock()];     
+        }
+        initGenesisBlock(){
+            return new BlockCrypto(0, "06/04/2021", "Initial Block in the Chain", "0");
+        }
+        latestBlock(){
+            return this.block1chain[this.block1chain.length - 1];
+        }
+        addNewBlock(newBlock){
+            newBlock.nextHash = this.latestBlock().hash;
+            newBlock.hash = newBlock.computeHash();        
+            this.block1chain.push(newBlock);
+        }
+    }
+```js
+
+
+Let's explain the above class 
+
+As usual. we have our constructor which instantiates the block1chain but this time we passed to it the `initGenesisBlock()` method which initializes the block in the chain. This property refers to an array of blocks in our case.
+
+- `initGenesisBlock()`: This is the first block created in the peer-to-peer network and has not been linked to any other. To our knowledge of indexing it's at index 0, Notice we created it using the previously created 'BlockCrypto' class and passed all the parameters.
+
+- `latestBlock` : As named, we can easily tell it's finding the latest block in the chain. As explained earlier it helps to ensure the hash of the current block and map it to the hash of the previous block to ensure chain integrity.
+
+- `addNewBlock`: This method I used here to add a new block to the chain. Matching the previous hash block and current hash block in our case is to ensure we are not tampering with the chain.
+
+
+Now that our blockchain is ready to work. We are missing something which is a core blockchain principle, the blockchain _integrity_. Let's see how to verify it and test our app.
```

Comment:
Why are we doing this? Explain the goal

Reply:
> Why are we doing this? Explain the goal
@srishilesh I'm trying to explain the code below there.